src/lib/work_partitioner/job_pool.ml

```diff
@@ -0,0 +1,125 @@
+open Core_kernel
+
+type ('accum, 'final) fold_action =
+  { action : [ `Continue of 'accum | `Stop of 'final ]; slashed : bool }
+
+let epoch_now () = Time.(now () |> to_span_since_epoch)
+
+module Make (ID : Hashtbl.Key) (Spec : T) = struct
```

Comment:
To confirm my understanding: `Snark_work_lib.Id` exposes two id types, and these will be exactly the types we'll use here.

Reply:
NIT: I'd suggest `Id` instead of `ID`, this is more inline with practice throughout the code